tt_metal/api/device.hpp

```diff
@@ -0,0 +1,342 @@
+ */
+const std::vector<uint32_t> &BankIdsFromLogicalCore(Device device, BufferType buffer_type, const CoreCoord &logical_core);
+
+/**
+ * @brief Deallocates all buffers on the device.
+ */
+void DeallocateBuffers(Device device);
+
+
+/**
+ * @brief Retrieves the machine epsilon for the SFPU on the device.
+ *
+ * @param device The device to query.
+ * @return SFPU machine epsilon.
+ */
+float GetSfpuEps(Device device);
+
+/**
+ * @brief Retrieves the representation of NaN for the SFPU on the device.
+ *
+ * @param device The device to query.
+ * @return SFPU NaN value.
+ */
+float GetSfpuNan(Device device);
+
+/**
+ * @brief Retrieves the representation of infinity for the SFPU on the device.
+ *
+ * @param device The device to query.
+ * @return SFPU infinity value.
+ */
+float GetSfpuInf(Device device);
+
+/**
+ * @brief Retrieves a command queue for the given device and queue ID.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID (default: 0).
+ * @return CommandQueue handle.
+ */
+CommandQueue GetCommandQueue(Device device, size_t cq_id = 0);
+
+/**
+ * @brief Begins tracing for a specific command queue ID and trace ID on the device.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID.
+ * @param tid The trace ID.
+ */
+void BeginTrace(Device device, uint8_t cq_id, uint32_t tid);
+
+/**
+ * @brief Ends tracing for a specific command queue ID and trace ID on the device.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID.
+ * @param tid The trace ID.
+ */
+void EndTrace(Device device, uint8_t cq_id, uint32_t tid);
+
+/**
+ * @brief Replays a trace for a specific command queue ID and trace ID on the device.
+ *
+ * @param device The device to query.
+ * @param cq_id The command queue ID.
+ * @param tid The trace ID.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void ReplayTrace(Device device, uint8_t cq_id, uint32_t tid, bool blocking);
+
+/**
+ * @brief Releases resources associated with a specific trace ID on the device.
+ *
+ * @param device The device to query.
+ * @param tid The trace ID.
+ */
+void ReleaseTrace(Device device, uint32_t tid);
+
+/**
+ * @brief Pushes a work function onto the device's work queue.
+ *
+ * @param device The device to which the work will be pushed.
+ * @param work The work function to execute.
+ * @param blocking Indicates whether the operation should be blocking (default: false).
+ */
+void PushWork(Device device, std::function<void()> &&work, bool blocking = false);
+
+/**
+ * @brief Pushes a shared work function onto the device's work queue.
+ *
+ * @param device The device to which the work will be pushed.
+ * @param work Shared pointer to the work function to execute.
+ * @param blocking Indicates whether the operation should be blocking (default: false).
+ */
+void PushWork(Device device, std::shared_ptr<std::function<void()>> work, bool blocking = false);
```

Comment:
Not sure this overload should continue to exist, a `std::function` not wrapped in a `std::shared_ptr` should be sufficient since it already allocates type-erased functors (i.e lambdas with captures) rather than storing within the class layout.